src/Activities/ServiceCollectionExtensions.cs

```diff
@@ -39,6 +39,8 @@ public static IServiceCollection AddOmexActivitySource(this IServiceCollection s
 			serviceCollection.TryAddSingleton<ActivityMetricsSender>();
 			serviceCollection.TryAddSingleton<ActivityEventSender>();
 			serviceCollection.TryAddSingleton<IActivitiesEventSender, AggregatedActivitiesEventSender>();
+			serviceCollection.TryAddSingleton<IExtraActivityBaggageDimensions>(_ => new ExtraActivityBaggageDimensions());
```

Comment:
Given the use of TryAddSingleton here, is the assumption that anyone registering a IExtraActivityBaggageDimensions implementation does so before AddOmexActivitySource gets called, (the default only gets registered where another implementation doesn't already exist in the DI container)?
If so the documentation should be clear about the hard requirement of DI registration ordering.

Reply:
That's why I suggested using the IOptions pattern for configuring custom dimensions.